### PR TITLE
Rebase Motion 5

### PR DESCRIPTION
**Pull request by tkoeppe**
This PR contains the changes I made when I attempted to rebase the Motion-5 branch onto the previous four motions and previous editorial changes.

@burblebee, @jensmaurer, @opensdh : could you kindly review the technical changes here?

### CODE REVIEW

File: source/classes.tex [hunk 1]
```diff
@@ -6755,7 +6755,7 @@
 
 \pnum
 Access to the deallocation function is checked statically,
-even though a different one might actually be executed.
+even if a different one is actually be executed.
```
Comment (jensmaurer) [inline source/classes.tex:6758]: "is actually be" is broken

File: source/declarations.tex [hunk 2]
```diff
@@ -8090,7 +8090,8 @@
 \indextext{class!linkage specification}%
 A C language linkage is ignored
 in determining the language linkage of
-class members and the
+class members,
+friend functions with a trailing \grammarterm{requires-clause}, and
```
Comment (jensmaurer) [inline source/declarations.tex:8094]: the trailing "the" got dropped accidentally

File: source/templates.tex [hunk 3]
```diff
@@ -2161,7 +2162,12 @@
 A template declaration in which the name of the template is followed by
 a \grammarterm{template-argument-list} is
 a partial specialization of
-the template named in the declaration\iref{temp.class.spec}.
+the template named in the declaration\iref{temp.spec.partial}.
+\begin{note}
+A \grammarterm{template-id}
+cannot be the declared name of a primary template declaration.
+However, this syntax is allowed in partial specializations\iref{temp.spec.partial}.
+\end{note}
```
Comment (jensmaurer) [inline source/templates.tex:2170]: This note feels very much redundant with the normative text in front of it. Please drop.

File: source/templates.tex [hunk 4]
```diff
@@ -3129,23 +3135,21 @@
...
 \pnum
 \indextext{specialization!class template partial}%
-A partial specialization of a class template provides an alternative definition
+A partial specialization of a template provides an alternative definition
 of the template that is used instead of the primary definition when the
 arguments in a specialization match those given in the partial
-specialization\iref{temp.class.spec.match}.
-A declaration of the primary template shall precede any specialization of
+specialization\iref{temp.spec.partial.match}.
+A declaration of the primary template shall precede any partial specialization of
```
Comment (jensmaurer) [inline source/templates.tex:3142]: The addition of "partial" here is wrong (this also applies to explicit specializations).

**Review by jensmaurer — CHANGES_REQUESTED** [review]
Mostly good.

**Comment by jensmaurer** [comment]
I have pushed a typo fix to the parent branch "motions-2020-11-cwg-5-as_in_paper".

File: source/classes.tex (hunk 1 above)
Comment (tkoeppe) [reply source/classes.tex:6758]: That was my mistake when merging, fixed.

File: source/templates.tex (hunk 4 above)
Comment (tkoeppe) [reply source/templates.tex:3142]: Thanks, reverted.

File: source/declarations.tex (hunk 2 above)
Comment (tkoeppe) [reply source/declarations.tex:8094]: Yes, thanks, my mistake when merging.

File: source/templates.tex (hunk 3 above)
Comment (tkoeppe) [reply source/templates.tex:2170]: Done, thanks!

**Comment by tkoeppe** [comment]
> I have pushed two typo-level fixes to the parent branch "motions-2020-11-cwg-5-as_in_paper".

Thanks! Rebased and force-pushed, including all the fixes you requested above.

**Review by jensmaurer — APPROVED** [review]

**Pull request merged**